src/mlpack/tests/imputation_test.cpp

```diff
@@ -0,0 +1,242 @@
+  BOOST_REQUIRE_CLOSE(output(0, 1), 99.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(0, 2), 2.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(0, 3), 0.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 0), 5.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 1), 6.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 2), 0.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 3), 6.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 0), 9.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 1), 8.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 2), 4.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 3), 8.0, 1e-5);
+}
+
+/**
+ * Make sure a CSV is loaded correctly.
+ */
+BOOST_AUTO_TEST_CASE(MeanImputationTest)
+{
+  arma::mat input("3.0 0.0 2.0 0.0;"
+                  "5.0 6.0 0.0 6.0;"
+                  "9.0 8.0 4.0 8.0;");
+  arma::mat outputT; // assume input is transposed
+  arma::mat output;  // assume input is not transposed
+  double mappedValue = 0.0;
+
+  MeanImputation<double> imputer;
+
+  // transposed
+  imputer.Apply(input, outputT, mappedValue, 0/*dimension*/, true);
+
+  BOOST_REQUIRE_CLOSE(outputT(0, 0), 3.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(0, 1), 2.5, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(0, 2), 2.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(0, 3), 2.5, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(1, 0), 5.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(1, 1), 6.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(1, 2), 0.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(1, 3), 6.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(2, 0), 9.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(2, 1), 8.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(2, 2), 4.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(2, 3), 8.0, 1e-5);
+
+  // not transposed
+  imputer.Apply(input, output, mappedValue, 1/*dimension*/, false);
+
+  BOOST_REQUIRE_CLOSE(output(0, 0), 3.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(0, 1), 7.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(0, 2), 2.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(0, 3), 0.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 0), 5.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 1), 6.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 2), 0.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(1, 3), 6.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 0), 9.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 1), 8.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 2), 4.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(output(2, 3), 8.0, 1e-5);
+}
+
+/**
+ * Make sure a CSV is loaded correctly.
+ */
+BOOST_AUTO_TEST_CASE(MedianImputationTest)
+{
+  arma::mat input("3.0 0.0 2.0 0.0;"
+                  "5.0 6.0 0.0 6.0;"
+                  "9.0 8.0 4.0 8.0;");
+  arma::mat outputT; // assume input is transposed
+  arma::mat output;  // assume input is not transposed
+  double mappedValue = 0.0;
+
+  MedianImputation<double> imputer;
+
+  // transposed
+  imputer.Apply(input, outputT, mappedValue, 1/*dimension*/, true);
+
+  BOOST_REQUIRE_CLOSE(outputT(0, 0), 3.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(0, 1), 0.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(0, 2), 2.0, 1e-5);
+  BOOST_REQUIRE_CLOSE(outputT(0, 3), 0.0, 1e-5);
```

Comment:
the dimension given here is 1 (second dimension)
so dimension 0 is left as the same as input.